client/src/components/MarkdownPage.js

```diff
@@ -16,7 +19,34 @@ export const MarkdownPage = ({
   markdown,
   width = 'large'
 }) => {
-  const config = {
+  const sectionId = getHash()
+  const [offset, setOffset] = useState(0)
+
+  useEffect(() => {
+    const sections = document.querySelectorAll(
+      'ol > li > p:first-child > span:first-child'
+    )
+    // we can print this to generate a list of linkable section text node for the config
+    // const sectionNames = Array.from(sections).map((item) => item.textContent)
+    for (const section of sections) {
+      section.id = formatStringToIdName(section.textContent)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (sectionId) {
+      const target = document.querySelector(sectionId)
+      target.scrollIntoView()
+    }
+    // (hack) prevents the selected section and the site header from overlapping on page load
+    const timer = setTimeout(() => {
+      setOffset(getHash() ? 90 : 0)
+    }, 0)
+
+    return () => clearTimeout(timer)
+  }, [sectionId])
```

Comment:
I'd like to avoid dynamically adding a margin to handle this if possible. 

Instead of the timeout would we be able to do something like this? 

(Also, we should check that target exists so we dont get an error page if someone tries to go to a random hash, but this is low stakes more just for completeness)

```suggestion
  useEffect(() => {
    const target = sectionId ? document.querySelector(sectionId) : null
    if (target) {
      const headerOffet = -90
      const y = target.getBoundingClientRect().top + window.pageYOffset + headerOffset;
      window.scrollTo({top: y, behavior: 'smooth'})
    }
  }, [sectionId])
```

Reply:
> I'd like to avoid dynamically adding a margin to handle this if possible.
> Instead of the timeout would we be able to do something like this?

Generally your suggestion is a great approach for these type of scenarios. Unfortunately, I had to include this workaround (using timer) since using hyperlinks, as opposed to direct access or page refresh, results in inconsistent scroll position. 

Please see the following Vercel previews and test the following test cases: 
**Without the workaround:** [preview](https://scpca-portal-l2txhsim2-ccdl.vercel.app/demo)
**With the workaround:** [preview](https://scpca-portal-bbpt97tvs-ccdl.vercel.app/demo)

**Case 1:** Visit the page using the link in a demo page
**Case 2:** Refresh the page 
**Case 3:** Access the page by entering its URL in a browser's address bar

Please let me know your thought on this.

> ...we should check that target exists so we dont get an error page if someone tries to go to a random hash, but this is low stakes more just for completeness

I've included the check 👍